lib/SILOptimizer/Mandatory/Differentiation.cpp

```diff
@@ -838,36 +838,67 @@ class ADContext {
   SILPassManager &getPassManager() const { return passManager; }
   Lowering::TypeConverter &getTypeConverter() { return module.Types; }
 
-  SmallVectorImpl<DifferentiableFunctionInst *> &
-  getDifferentiableFunctionInsts() {
-    return differentiableFunctionInsts;
+  /// Returns true if there are `DifferentiableFunctionInst` to process.
+  bool hasDifferentiableFunctionInsts() const {
+    return !differentiableFunctionInsts.empty();
   }
 
-  SmallPtrSetImpl<DifferentiableFunctionInst *> &
-  getProcessedDifferentiableFunctionInsts() {
-    return processedDifferentiableFunctionInsts;
+  /// Pick and remove a `DifferentiableFunctionInst` for processing.
+  /// Returns nullptr if none are available for processing.
+  DifferentiableFunctionInst* pickAndRemoveDifferentiableFunctionInst() {
+    if (differentiableFunctionInsts.empty()) return nullptr;
+    auto *dfi = differentiableFunctionInsts.back();
+    differentiableFunctionInsts.pop_back();
+    return dfi;
+  }
+
+  void addDifferentiableFunctionInst(DifferentiableFunctionInst* dfi) {
+    differentiableFunctionInsts.push_back(dfi);
+  }
+
+  bool
+  isDifferentiableFunctionInstProcessed(DifferentiableFunctionInst *dfi) const {
+    return processedDifferentiableFunctionInsts.count(dfi) > 0;
```

Comment:
`count` is the name for `contains` in LLVM APIs. Here, it effectively returns a `bool`, so it's idiomatic to use its result as a `bool` directly.

```suggestion
    return processedDifferentiableFunctionInsts.count(dfi);
```